keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepositoryTests.kt

```diff
@@ -205,23 +205,24 @@ abstract class ResourceRepositoryTests<T : ResourceRepository> : JUnit5Minutests
                 .isA<ResourceValid>()
             }
 
-            context("a subsequent identical event that should be ignored") {
+            context("a subsequent identical event with duplicates disallowed") {
               before {
                 tick()
                 subject.appendHistory(ResourceValid(resource, clock))
               }
 
-              test("the event is not included in the history") {
+              test("the existing event's timestamp is updated") {
                 expectThat(subject.eventHistory(resource.id))
                   .and {
                     first().isA<ResourceValid>()
+                      .and { get { timestamp }.isEqualTo(clock.instant()) }
```

Comment:
This is failing, and I don't understand why... 😭 
The update is logged as expected:
```
-> with bind values      : update `keel`.`event` set `keel`.`event`.`timestamp` = {ts '2020-05-09 00:40:48.588'} where (`keel`.`event`.`scope` = 'RESOURCE' and `keel`.`event`.`uid` = '01E7VE1BQMRY0KET3XPYHDH1WH') order by `keel`.`event`.`timestamp` desc limit 1
Affected row(s)          : 1
```
But the test still fails:
```
org.opentest4j.AssertionFailedError: ▼ Expect that [ResourceValid(kind=test/whatever@v1, id=test:whatever:37396637, application=fnord, timestamp=2020-05-09T00:40:47.588Z), ResourceUpdated(kind=test/whatever@v1, id=test:whatever:37396637, application=fnord, delta={}, timestamp=2020-05-09T00:40:46.588Z), ResourceCreated(kind=test/whatever@v1, id=test:whatever:37396637, application=fnord, timestamp=2020-05-09T00:40:45.588Z)]:
  ▼ first element ResourceValid(kind=test/whatever@v1, id=test:whatever:37396637, application=fnord, timestamp=2020-05-09T00:40:47.588Z):
    ✓ is an instance of com.netflix.spinnaker.keel.events.ResourceValid
    ▼ 2020-05-09T00:40:47.588Z:
      ✗ is equal to 2020-05-09T00:40:48.588Z
              found 2020-05-09T00:40:47.588Z
```
What am I missing?!

Reply:
What I was missing was to update the redundant `timestamp` field within the JSON... 😛 We should really remove that redundancy, it's a recipe for bugs.